QSB/Events/FullStateMessage.cs

```diff
@@ -0,0 +1,38 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using QSB.Messaging;
+using UnityEngine.Networking;
+
+namespace QSB.Events
+{
+    class FullStateMessage : NameMessage
+    {
+        public override MessageType MessageType => MessageType.FullState;
+
+        public Dictionary<uint, string> PlayerNames;
+
+        public override void Deserialize(NetworkReader reader)
+        {
+            base.Deserialize(reader);
+            PlayerNames = new Dictionary<uint, string>();
+
+            reader.ReadString().Split(',').ToList().ForEach(pair =>
+            {
+                var splitPair = pair.Split(':');
+                var key = Convert.ToUInt16(splitPair[0]);
+                var value = splitPair[1];
+                PlayerNames[key] = value;
+            });
```

Comment:
This method should only return the deserialized object, not change stuff. I suggest updating PlayerNames is done by the "user" of the message instead.